doc: clarify conversions can be impacted by double-rounding (#2361)
oneapi-src/oneDNN: doc/programming_model/data_types.md

the floating-point math controls described in @ref
dev_guide_attributes_fpmath_mode.

> The \f$convert_{dst\_dt}\f$ conversion is not guaranteed to be correctly

Reviewer: Will this be the result of the conversion is not guaranteed to be correctly rounded to the nearest value?

Author: I clarified that it would be faithfully rounded, and what that means (one of the two closest representable values is returned, but not necessarily the closest).

rounded. In particular, some hardware platforms have no direct

> conversion instructions from f32 to low-precision datatypes (e.g. fp8

Reviewer: ```suggestion
conversion instructions to convert f32 data types to low-precision data types such as fp8
```

Author: fixed, thanks

> or fp4), and will perform conversion through an intermediate datatype

Reviewer: ```suggestion
or fp4, and will perform conversion through an intermediate data type
```

Author: fixed thanks

> (e.g. f16/bf16), which may cause [double

Reviewer: ```suggestion
(for example, f16 or bf16), which may result in [double
```

Author: fixed, thanks.

rounding](https://en.wikipedia.org/wiki/Rounding#Double_rounding).

> ### Rounding mode and denormal handling

Reviewer 2: Does it make sense to place this note to [rounding mode](https://github.com/oneapi-src/oneDNN/blob/main/doc/programming_model/data_types.md#rounding-mode-and-denormal-handling) section instead?

Author: That is an option, but given that it is not about rounding mode but rounding accuracy, I thought it might be better to put with the general formula.

Reviewer 2: Got it, thanks.
